commit 2a68a6cd3ce8c24a18773da888411d0f135798c6
Author: Richard Hughes <richard@hughsie.com>
Date:   Mon Sep 13 15:32:42 2010 +0100

    browser-plugin: Do not query PkDesktop by default
    
    By default, firefox is compiled with an internal version of sqlite
    and when packagekit-glib sucks its own copy into the running
    process, hilarity ensues. Icons are less important than a crashed
    browser.

diff --git a/contrib/browser-plugin/pk-plugin-install.c b/contrib/browser-plugin/pk-plugin-install.c
index f940639..f854f28 100644
--- a/contrib/browser-plugin/pk-plugin-install.c
+++ b/contrib/browser-plugin/pk-plugin-install.c
@@ -38,6 +38,8 @@
 #include "pk-main.h"
 #include "pk-plugin-install.h"
 
+//#define PK_PLUGIN_INSTALL_USE_DESKTOP_FOR_INSTALLED
+
 #define PK_PLUGIN_INSTALL_GET_PRIVATE(o) (G_TYPE_INSTANCE_GET_PRIVATE ((o), PK_TYPE_PLUGIN_INSTALL, PkPluginInstallPrivate))
 
 typedef enum {
@@ -189,11 +191,12 @@ pk_plugin_install_set_installed_version (PkPluginInstall *self, const gchar *ver
 static gchar *
 pk_plugin_install_get_best_desktop_file (PkPluginInstall *self)
 {
+	gchar *data = NULL;
+#ifdef PK_PLUGIN_INSTALL_USE_DESKTOP_FOR_INSTALLED
 	GPtrArray *array = NULL;
 	PkDesktop *desktop;
 	GError *error = NULL;
 	gboolean ret;
-	gchar *data = NULL;
 	const gchar *package;
 
 	/* open desktop database */
@@ -230,6 +233,7 @@ out:
 		g_ptr_array_free (array, TRUE);
 	}
 	g_object_unref (desktop);
+#endif
 	return data;
 }
 
@@ -328,9 +332,7 @@ pk_plugin_install_finished_cb (GObject *object, GAsyncResult *res, PkPluginInsta
 #endif
 		}
 
-		if (self->priv->app_info != 0)
-			pk_plugin_install_set_status (self, INSTALLED);
-
+		pk_plugin_install_set_status (self, INSTALLED);
 		pk_plugin_install_clear_layout (self);
 		pk_plugin_install_refresh (self);
 	}
